### PR TITLE
Support dot notation with `$unset`

### LINKED ISSUE #153: Support nested path ("dot notation") for `$unset` operator
(note: sub-task of #74 )

Although there are common pieces wrt "dot notation" paths for update operations, there are some differences so I'll first tackle simplest case of `$unset` and probably create separate issues for other operators too -- this helps keep PRs from growing too big.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds support for dotted notation for `$unset` operation, along with tests

**Which issue(s) this PR fixes**:
Fixes #153

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/UpdateTarget.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Definition of a target for an {@link UpdateOperation}; built from "dot path" and document to
+ * update, by {@link UpdateTargetLocator}.
+ */
+public record UpdateTarget(
+    String fullPath, JsonNode contextNode, JsonNode valueNode, String lastProperty, int lastIndex) {
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/UpdateTarget.java:12]: Can you add java doc not able to follow what is lastProperty and lastIndex for?

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/SetOperation.java [hunk 2]
```diff
@@ -28,7 +28,7 @@ public static SetOperation construct(ObjectNode args) {
   }
 
   @Override
-  public boolean updateDocument(ObjectNode doc) {
+  public boolean updateDocument(ObjectNode doc, UpdateTargetLocator targetLocator) {
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/SetOperation.java:31]: Set operation also need to support dot notation, Is this PR only for unset?
Eg: 
`db.products.updateOne(
{ _id: 100 },
{ $set: { "details.make": "Kustom Kidz" } }
)`

**Review by maheshrajamani — APPROVED** [review]
Is this PR only limited to unset operation, we need to support dot notation for other operation update too. Will we be doing it as separate PR?

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindAndUpdateIntegrationTest.java [hunk 3]
```diff
@@ -415,6 +402,91 @@ public void findByIdAndUnset() {
...
+      //
+      // * array.0   -> remove first entry, replace with null
+      // * array.1.x -> remove property 'x' from second array element (object)
+      // * subdoc.y  -> remove subdoc property 'y'
+      String json =
+          """
+                   {
+                      "findOneAndUpdate": {
+                        "filter" : {"_id" : "update_doc_unset_nested"},
+                        "update" : {"$unset" : {
```
Comment (vkarpov15) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindAndUpdateIntegrationTest.java:437]: Would also be nice to test that `$unset: { _id: 1 }` throws an error

**Review by vkarpov15 — APPROVED** [review]
Quick test comment, but otherwise LGTM

**Comment by tatu-at-datastax** [comment]
> Is this PR only limited to unset operation, we need to support dot notation for other operation update too. Will we be doing it as separate PR?

Yes, I just wanted to keep the first PR more limited -- I will next add this to `$set`, and then maybe last ones as a whole.

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/UpdateTarget.java (hunk 1 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/UpdateTarget.java:12]: Will do.

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/SetOperation.java (hunk 2 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/SetOperation.java:31]: Correct: this is only for $unset, #166 is the issue for $set and I'll work on that next.

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindAndUpdateIntegrationTest.java (hunk 3 above)
Comment (tatu-at-datastax) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindAndUpdateIntegrationTest.java:437]: I think that is already tested in another test (unit test I think). But I can add one here just for completeness' sake.

**Comment by tatu-at-datastax** [comment]
Added Javadocs to explain more; added `_id` + `$unset` fail IT. Merging.

**Pull request merged**